### ISSUE TITLE
[BUG] [Form Recognizer SDK] Unable to serialize AnalyzeResult to JSON with 4.0.0 Version Release

### ISSUE THREAD

**Issue by agsanderss** [Azure/azure-sdk-for-java#31172]
**Describe the bug**
The Form Recognizer AnalyzeResult for prebuilt models was able to be parsed as JSON using Jackson or GSON libraries to return the result to a client as an API response with 4.0.0-beta.5 or 6 versions. 

When upgrading to the 4.0.0 release, we are receiving a JSON Parse error with the new object structure.

***Exception or Stack Trace***
class java.lang.String cannot be cast to class java.lang.Double (java.lang.String and java.lang.Double are in module java.base of loader 'bootstrap') (through reference chain: com.azure.ai.formrecognizer.documentanalysis.models.AnalyzeResult["documents"]->java.util.ArrayList[0]->com.azure.ai.formrecognizer.documentanalysis.models.AnalyzedDocument["fields"]->java.util.HashMap["Dependents"]->com.azure.ai.formrecognizer.documentanalysis.models.DocumentField["value"]->java.util.ArrayList[0]->com.azure.ai.formrecognizer.documentanalysis.models.DocumentField["value"]->java.util.HashMap["Name"]->com.azure.ai.formrecognizer.documentanalysis.models.DocumentField["valueAsDouble"])

**To Reproduce**
1. Use 4.0.0 version of the azure.ai.formrecognizer sdk
2. Select an image to use with "prebuilt-healthInsuranceCard.us" model
3. Use com.fasterxml.jackson.core:jackson-databind:2.13.4 ObjectMapper
4. Retrieve the Analyze Result and attempt to write it to JSON 

***Code Snippet***

           DocumentAnalysisClient client = new DocumentAnalysisClientBuilder()
                .credential(creds)
                .endpoint(endpoint)
                .buildClient();

            String modelId = "prebuilt-healthInsuranceCard.us";
            SyncPoller <OperationResult, AnalyzeResult> analyzeDocumentPoller = client.beginAnalyzeDocument(modelId, data);
            AnalyzeResult analyzeLayoutResult = analyzeDocumentPoller.getFinalResult();

            ObjectMapper obj = new ObjectMapper();
            System.out.println(obj.writeValueAsString(analyzeLayoutResult));

**Expected behavior**
AnalyzeResult object should be able to be parsed as JSON

**Setup (please complete the following information):**
 - OS: Windows 11
 - IDE: VSCode
 - Library/Libraries: com.azure:azure-ai-formrecognizer:4.0.0
 - Java version: 8
 - App Server/Environment: N/A
 - Frameworks: N/A

**Additional context**
This worked in previous versions of the Beta Libraries and any of the libraries before 4.0.0

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

**Comment by joshfree** [comment]
Hi @agsanderss, thank you for reporting this issue.  @samvaity can you please follow up?

**Comment by samvaity** [comment]
Hi @agsanderss, thanks for reporting. Could you provide the `"data"` you are analyzing to reproduce the error?

**Comment by agsanderss** [comment]
Hi @samvaity - I'm not able to provide the exact image, but it's a jpg file of an insurance card. Adding a code snippet here of what's getting passed in: 

```
File insuranceCard = new File(filePath);
            byte[] bytes = Files.readAllBytes(insuranceCard.toPath());
            BinaryData data = BinaryData.fromBytes(bytes);
```

**Comment by samvaity** [comment]
Hey @agsanderss, I am able to successfully retrieve data on analyzing [insurance card](https://raw.githubusercontent.com/Azure/azure-sdk-for-js/main/sdk/formrecognizer/ai-form-recognizer/assets/healthInsuranceCard/insurance.png) for `azure-ai-formrecognier 4.0.0` .
Can you try with the provided example of insurance card and see if it works for you?

**Comment by agsanderss** [comment]
Hey @samvaity - the issue doesn't occur on retrieving the result. The issue occurs when attempting to serialize the result as JSON using gson or jackson libraries.

In azure-ai-formrecognier 4.0.0-beta.6 we can serialize the AnalyzeResult object into JSON for usage of our downstream consumers, in 4.0.0 this fails which forces us to write custom code to handle the nested hashmap that gets returned. Our usage pattern is to send back exactly what Form Recognizer returns to consumers rather than parsing out specific fields.

With 4.0.0, this is the code that isn't working:
`String modelId = "prebuilt-healthInsuranceCard.us";
SyncPoller <OperationResult, AnalyzeResult> analyzeDocumentPoller = client.beginAnalyzeDocument(modelId, data);
AnalyzeResult analyzeLayoutResult = analyzeDocumentPoller.getFinalResult();

ObjectMapper obj = new ObjectMapper();
System.out.println(obj.writeValueAsString(analyzeLayoutResult)); //This is what fails `

With 4.0.0-beta.6, this code works:
`String modelId = "prebuilt-healthInsuranceCard.us";
SyncPoller <DocumentOperationResult, AnalyzeResult> analyzeDocumentPoller = client.beginAnalyzeDocument(modelId, data, data.getLength());
AnalyzeResult analyzeLayoutResult = analyzeDocumentPoller.getFinalResult();

ObjectMapper obj = new ObjectMapper();
System.out.println(obj.writeValueAsString(analyzeLayoutResult));`

Let me know if there's anything else I can share to help with this.

**Comment by samvaity** [comment]
Hey @agsanderss, thanks for the additional information. I am able to reproduce the error and working on a fix right now.

**Comment by samvaity** [comment]
Hi @agsanderss, the fix was merged and released as a part of [azure-ai-formrecognizer/4.0.1](https://mvnrepository.com/artifact/com.azure/azure-ai-formrecognizer/4.0.1).
Let us know if this issue still exists.
